Fix broken Travis script

I think this requires an adjustment to the `.travis.yml`., perhaps based on https://github.com/quanteda/readtext/pull/133.

I'd be interested in a fix for this one too, since I'd like to be able to use have readtext as a dependency. For me it's an `ndjson` installation failure (below) that stops `streamR` installation which stops `readtext` installation.  (Paul's two apt-get fixes are already in).

```
'/home/travis/R-bin/lib/R/bin/R' --no-site-file --no-environ --no-save  \
  --no-restore --quiet CMD INSTALL  \
  '/tmp/Rtmp3woujk/devtools387c2fb58294/ndjson'  \
  --library='/home/travis/R/Library' --install-tests 

* installing *source* package â€˜ndjsonâ€™ ...
** package â€˜ndjsonâ€™ successfully unpacked and MD5 sums checked
** libs
g++ -std=gnu++11 -I"/home/travis/R-bin/lib/R/include" -DNDEBUG  -I"/home/travis/R/Library/Rcpp/include" -I/home/travis/R-bin/include   -fpic  -g -O2 -c RcppExports.cpp -o RcppExports.o
g++ -std=gnu++11 -I"/home/travis/R-bin/lib/R/include" -DNDEBUG  -I"/home/travis/R/Library/Rcpp/include" -I/home/travis/R-bin/include   -fpic  -g -O2 -c gzstream.cpp -o gzstream.o
g++ -std=gnu++11 -I"/home/travis/R-bin/lib/R/include" -DNDEBUG  -I"/home/travis/R/Library/Rcpp/include" -I/home/travis/R-bin/include   -fpic  -g -O2 -c ndjson.cpp -o ndjson.o
In file included from ndjson.cpp:8:0:
json.h:127:10: error: #error "unsupported GCC version - see https://github.com/nlohmann/json#supported-compilers"
         #error "unsupported GCC version - see https://github.com/nlohmann/json#supported-compilers"
          ^
make: *** [ndjson.o] Error 1
ERROR: compilation failed for package â€˜ndjsonâ€™
* removing â€˜/home/travis/R/Library/ndjsonâ€™
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -6,18 +6,20 @@ warnings_are_errors: true
 sudo: true
 dist: trusty
 
-script:
-  - sudo unlink /usr/bin/gcc && sudo ln -s /usr/bin/gcc-5 /usr/bin/gcc
-  - gcc --version
 addons:
   apt:
-    sources:
-      - ubuntu-toolchain-r-test
     packages:
-      - gcc-5
-      - g++-5
       - libapparmor-dev
       - libpoppler-cpp-dev
+
+before_script:
+  - sudo add-apt-repository ppa:ubuntu-toolchain-r/test
+  - sudo apt-get update
+  - sudo apt-get install gcc-5
+  - sudo apt-get install g++-5
+  - sudo unlink /usr/bin/gcc && sudo ln -s /usr/bin/gcc-5 /usr/bin/gcc
+  - gcc --version
+
 
 env:
     global:

--- a/.travis.yml
+++ b/.travis.yml
@@ -15,15 +15,15 @@ addons:
 before_install:
   - sudo add-apt-repository ppa:ubuntu-toolchain-r/test -y
   - sudo apt-get update
-  - sudo apt-get install gcc-6
-  - sudo unlink /usr/bin/gcc && sudo ln -s /usr/bin/gcc-6 /usr/bin/gcc
+  - sudo apt-get install gcc-5
+  - sudo unlink /usr/bin/gcc && sudo ln -s /usr/bin/gcc-5 /usr/bin/gcc
   #- sudo update-alternatives --query gcc
   - gcc --version
-  - sudo apt-get install g++-6 
-  - sudo unlink /usr/bin/g++ && sudo ln -s /usr/bin/g++-6 /usr/bin/g++
+  - sudo apt-get install g++-5
+  - sudo unlink /usr/bin/g++ && sudo ln -s /usr/bin/g++-5 /usr/bin/g++
   - g++ --version
-  - sudo apt-get install gfortran-6
-  - sudo unlink /usr/bin/gfortran && sudo ln -s /usr/bin/gfortran-6 /usr/bin/gfortran
+  - sudo apt-get install gfortran-5
+  - sudo unlink /usr/bin/gfortran && sudo ln -s /usr/bin/gfortran-5 /usr/bin/gfortran
   - gfortran --version
 
 

--- a/.travis.yml
+++ b/.travis.yml
@@ -13,7 +13,7 @@ addons:
       - libpoppler-cpp-dev
 
 before_install:
-  - sudo add-apt-repository ppa:ubuntu-toolchain-r/test
+  - sudo add-apt-repository ppa:ubuntu-toolchain-r/test -y
   - sudo apt-get update
   - sudo apt-get install gcc-5
   - sudo apt-get install g++-5

--- a/.travis.yml
+++ b/.travis.yml
@@ -10,6 +10,11 @@ addons:
     packages:
       - libapparmor-dev
       - libpoppler-cpp-dev
+before_install:
+  #https://gist.github.com/cotsog/3ce84675af0d74438d91
+  - sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
+  - sudo apt-get update -qq
+
 env:
     global:
     - R_CHECK_ARGS="--no-build-vignettes"

--- a/.travis.yml
+++ b/.travis.yml
@@ -5,15 +5,19 @@ language: r
 warnings_are_errors: true
 sudo: true
 dist: trusty
+
+script:
+  - sudo unlink /usr/bin/gcc && sudo ln -s /usr/bin/gcc-5 /usr/bin/gcc
+  - gcc --version
 addons:
   apt:
+    sources:
+      - ubuntu-toolchain-r-test
     packages:
+      - gcc-5
+      - g++-5
       - libapparmor-dev
       - libpoppler-cpp-dev
-before_install:
-  #https://gist.github.com/cotsog/3ce84675af0d74438d91
-  - sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
-  - sudo apt-get update -qq
 
 env:
     global:

--- a/.travis.yml
+++ b/.travis.yml
@@ -16,9 +16,11 @@ before_install:
   - sudo add-apt-repository ppa:ubuntu-toolchain-r/test -y
   - sudo apt-get update
   - sudo apt-get install gcc-5
-  - sudo apt-get install g++-5
   - sudo unlink /usr/bin/gcc && sudo ln -s /usr/bin/gcc-5 /usr/bin/gcc
   - gcc --version
+  - sudo apt-get install g++-5
+  - sudo unlink /usr/bin/g++ && sudo ln -s /usr/bin/g++-5 /usr/bin/g++
+  - g++ --version
 
 
 env:

--- a/.travis.yml
+++ b/.travis.yml
@@ -15,7 +15,7 @@ addons:
 before_install:
   - sudo add-apt-repository ppa:ubuntu-toolchain-r/test -y
   - sudo apt-get update
-  - sudo apt-get install gcc-6 
+  - sudo apt-get install gcc-6
   - sudo unlink /usr/bin/gcc && sudo ln -s /usr/bin/gcc-6 /usr/bin/gcc
   #- sudo update-alternatives --query gcc
   - gcc --version

--- a/.travis.yml
+++ b/.travis.yml
@@ -23,8 +23,8 @@ before_install:
   - sudo unlink /usr/bin/g++ && sudo ln -s /usr/bin/g++-6 /usr/bin/g++
   - g++ --version
   - sudo apt-get install gfortran-6
-  - sudo unlink /usr/bin/gfortrun && sudo ln -s /usr/bin/gfortrun-6 /usr/bin/gfortrun
-  - gfortrun --version
+  - sudo unlink /usr/bin/gfortran && sudo ln -s /usr/bin/gfortran-6 /usr/bin/gfortran
+  - gfortran --version
 
 
 env:

--- a/.travis.yml
+++ b/.travis.yml
@@ -22,6 +22,9 @@ before_install:
   - sudo apt-get install g++-6 
   - sudo unlink /usr/bin/g++ && sudo ln -s /usr/bin/g++-6 /usr/bin/g++
   - g++ --version
+  - sudo apt-get install gfortran-6
+  - sudo unlink /usr/bin/gfortrun && sudo ln -s /usr/bin/gfortrun-6 /usr/bin/gfortrun
+  - gfortrun --version
 
 
 env:

--- a/.travis.yml
+++ b/.travis.yml
@@ -12,7 +12,7 @@ addons:
       - libapparmor-dev
       - libpoppler-cpp-dev
 
-before_script:
+before_install:
   - sudo add-apt-repository ppa:ubuntu-toolchain-r/test
   - sudo apt-get update
   - sudo apt-get install gcc-5

--- a/.travis.yml
+++ b/.travis.yml
@@ -15,11 +15,13 @@ addons:
 before_install:
   - sudo add-apt-repository ppa:ubuntu-toolchain-r/test -y
   - sudo apt-get update
-  - sudo apt-get install gcc-5
-  - sudo unlink /usr/bin/gcc && sudo ln -s /usr/bin/gcc-5 /usr/bin/gcc
+  - sudo update-alternatives --query gcc
+  - sudo apt-get install gcc-6
+  - sudo unlink /usr/bin/gcc && sudo ln -s /usr/bin/gcc-6 /usr/bin/gcc
+  - sudo update-alternatives --query gcc
   - gcc --version
-  - sudo apt-get install g++-5
-  - sudo unlink /usr/bin/g++ && sudo ln -s /usr/bin/g++-5 /usr/bin/g++
+  - sudo apt-get install g++-6
+  - sudo unlink /usr/bin/g++ && sudo ln -s /usr/bin/g++-6 /usr/bin/g++
   - g++ --version
 
 

--- a/.travis.yml
+++ b/.travis.yml
@@ -15,11 +15,11 @@ addons:
 before_install:
   - sudo add-apt-repository ppa:ubuntu-toolchain-r/test -y
   - sudo apt-get update
-  - sudo apt-get install gcc-6
+  - sudo apt-get install gcc-6 
   - sudo unlink /usr/bin/gcc && sudo ln -s /usr/bin/gcc-6 /usr/bin/gcc
-  - sudo update-alternatives --query gcc
+  #- sudo update-alternatives --query gcc
   - gcc --version
-  - sudo apt-get install g++-6
+  - sudo apt-get install g++-6 
   - sudo unlink /usr/bin/g++ && sudo ln -s /usr/bin/g++-6 /usr/bin/g++
   - g++ --version
 

--- a/.travis.yml
+++ b/.travis.yml
@@ -15,7 +15,6 @@ addons:
 before_install:
   - sudo add-apt-repository ppa:ubuntu-toolchain-r/test -y
   - sudo apt-get update
-  - sudo update-alternatives --query gcc
   - sudo apt-get install gcc-6
   - sudo unlink /usr/bin/gcc && sudo ln -s /usr/bin/gcc-6 /usr/bin/gcc
   - sudo update-alternatives --query gcc

